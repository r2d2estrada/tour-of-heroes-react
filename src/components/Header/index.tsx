import React from "react";
import { Navbar, Nav, Container } from "react-bootstrap";
import { NavLink, Link } from "react-router-dom";
import logo from "./../../assets/img/logo192.png";

const styles = {
  img: {
    maxHeight: 50,
    marginRight: "1rem",
  },
};

const Header: React.FC = () => {
  return (
    <Navbar bg="dark" variant="dark" expand="lg">
      <Container>
        <Navbar.Brand as={Link} to="/">
          <img
            src={logo}
            title="React Tour of Heroes"
            alt="Logo"
            style={styles.img}
          />
          Tour of Heroes React
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            <Nav.Link as={NavLink} exact to="/">
              Dashboard
            </Nav.Link>
            <Nav.Link as={NavLink} to="/heroes">
              Heroes
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
