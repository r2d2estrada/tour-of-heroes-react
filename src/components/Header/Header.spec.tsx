import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import Header from ".";

describe("Header", () => {
  let component: any;

  beforeAll(() => {
    component = () => {
      const { container } = render(
        <Router>
          <Header />
        </Router>
      );
      return container;
    };
  });

  it("should match snapshot", () => {
    expect(component()).toMatchSnapshot();
  });
});
