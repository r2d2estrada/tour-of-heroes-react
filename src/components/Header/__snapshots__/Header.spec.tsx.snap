// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt="Logo"
          src="logo192.png"
          style="max-height: 50px; margin-right: 1rem;"
          title="React Tour of Heroes"
        />
        Tour of Heroes React
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
        >
          <a
            aria-current="page"
            class="nav-link active"
            href="/"
          >
            Dashboard
          </a>
          <a
            class="nav-link"
            href="/heroes"
          >
            Heroes
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
